fw/http.c

```diff
@@ -1600,12 +1601,12 @@ static bool
 tfw_http_hm_suspend(TfwHttpResp *resp, TfwServer *srv)
 {
 	unsigned long old_flags, flags = READ_ONCE(srv->flags);
+	bool hm = flags & TFW_SRV_F_HMONITOR;
 
-	if (!(flags & TFW_SRV_F_HMONITOR))
-		return true;
-
-	if (!tfw_apm_hm_srv_limit(resp->status, srv->apmref))
+	if (!tfw_apm_hm_srv_limit(resp->status, srv->apmref) && hm)
```

Comment:
I'm not sure why this change is required. What is the difference? Why is there the need to check `hm` twice? Also, `hm` was introduced just for this purpose.

Reply:
I introduced the total response statistics in the same data structure, that serves limit calculation (`TfwApmHMStats`). Also that's what I tried to describe in the comments in `fw/apm.h`. That was optimal from the point of view of data structures, but naming probably suffered, but I didn't refactor it.

Reply:
So now we need to `tfw_apm_hm_srv_limit` to be called even if health monitoring is disabled, just to count statistics.

Reply:
Thanks. What about an alternative that clearly shows the logic that the function is called unconditionally? Something along these lines:
```C
r = tfw_apm_hm_srv_limit(resp->status, srv->apmref);
if ((!(flags & TFW_SRV_F_HMONITOR))
    return true;
if (!r)
    return false.
```

Reply:
Also, I think the comment that it's called unconditionally because of this and that would do good here.

Reply:
Done, thanks!